src/kOS.Safe/Encapsulation/EnumerableValue.cs

```diff
@@ -1,25 +1,27 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
-using System.Linq;
 using kOS.Safe.Encapsulation.Suffixes;
 using kOS.Safe.Serialization;
+using System.Linq;
 
 namespace kOS.Safe.Encapsulation
 {
-    public abstract class EnumerableValue<T, TC> : SerializableStructure, IEnumerable<T> where TC : IEnumerable<T> where T : Structure
+    public abstract class EnumerableValue<T, TE> : SerializableStructure, IEnumerable<T> where TE : IEnumerable<T> where T : Structure
     {
-        protected TC Collection { get; private set; }
+        protected TE Enum { get; private set; }
```

Comment:
It looks like this was changed to InnerEnum in your subclasses, but not here in EnumerableValue.  I'll correct the issue with a refactor.